src/main/java/connor/task/TaskList.java

```diff
@@ -0,0 +1,269 @@
+package connor.task;
+
+import connor.Connor;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+
+import java.util.ArrayList;
+
+/**
+ * Stores and edits the task list.
+ */
+public class TaskList {
+    private static final String INDENT = Connor.INDENT;
+    private static final String LINE = Connor.LINE;
+    private static final String EMPTY_TASK_LIST = "Your task list is empty.";
+    private static final String SHOW_TASKS = "Here are your current tasks: ";
+    private static final String ADD_NEW_TASK = "Alright, I've added a new task: ";
+    private static final String DELETE_TASK = "Alright, I've deleted the task: ";
+    private static final String CLEAR_TASKS_CONFIRM = "Are you REALLY sure you want to clear all your tasks?\n"
+            + "Type 'yes' to confirm, type anything else (or nothing) to cancel.";
+    private static final String CLEAR_TASKS_CONFIRMED = "Poof! All your tasks are cleared!";
+    private static final String CLEAR_TASKS_CANCEL = "Cancelled clearing all tasks. Phew!";
+    private static final String MARK_TASK = "Good job! I've marked the following task as completed: ";
+    private static final String UNMARK_TASK = "Understood. I've unmarked the following task: ";
+    private static final String NO_MATCHING_TASKS = "Sorry, there are no matching tasks in your list.";
+    private static final String SHOW_MATCHING_TASKS = "Here are the matching tasks in your list: ";
+    private static final String ERROR_EMPTY_INDEX = "Error! Index cannot be empty.";
+    private static final String ERROR_EMPTY_TASK_DESC = "Error! Tasks cannot have an empty description.";
+    private static final String ERROR_EMPTY_DL_DESC = "Error! Deadlines cannot have empty descriptions or dates.";
+    private static final String ERROR_EMPTY_EVENT_DESC = "Error! Events cannot have empty descriptions or dates.";
+    private static final String ERROR_INDEX_NOT_INTEGER = "Error! Index must be a valid integer.";
+    private static final String ERROR_INDEX_OUT_OF_RANGE = "Error! Given index is out of range.";
+    private static final String ERROR_INVALID_DL_FORMAT = "Error! Wrong format for deadlines.\n\n"
+            + "Deadline tasks must include \"by\" in the description.\nExample: \n"
+            + ">>> deadline Finish project /by Monday Morning";
+    private static final String ERROR_INVALID_EVENT_FORMAT = "Error! Wrong format for events.\n\n"
+            + "Event tasks must include \"at\" in the description.\nExample: \n"
+            + ">>> event Birthday Party at May 5th";
+    private static final String ERROR_INVALID_TASK_TYPE = "Oh no! Incorrect Task type!";
+    private static final String ERROR_MARK_EMPTY = "Error! I can't mark an empty task list!";
+    private static final String ERROR_UNMARK_EMPTY = "Error! I can't unmark an empty task list!";

```

Comment:
I like that you encapsulate every error message into one file, to make it easier to edit them in the future.

Reply:
Thank you! I agree, it is very convenient, especially if the message appears multiple times.